lib/partition_healing/discover_provider_healer.js

```diff
@@ -43,10 +44,63 @@ var Errors = {
  */
 function DiscoverProviderHealer(ringpop) {
     DiscoverProviderHealer.super_.call(this, ringpop);
+    this.timers = ringpop.timers || globalTimers;
+
     this.maxNumberOfFailures = ringpop.config.get('discoverProviderHealerMaxFailures');
+    this.healPeriod = ringpop.config.get('discoverProviderHealerPeriod');
+    this.baseProbability = ringpop.config.get('discoverProviderHealerBaseProbability');
+
+    this.previousHostListSize = 0;
+    this.healTimer = null;
+    this.isStopped = true;
 }
 require('util').inherits(DiscoverProviderHealer, Healer);
 
+DiscoverProviderHealer.prototype.start = function start() {
+    if (this.healTimer) {
+        return;
+    }
+    this.isStopped = false;
+    this._run();
+};
+
+DiscoverProviderHealer.prototype.stop = function stop() {
+    this.isStopped = true;
+    if (this.healTimer) {
+        this.timers.clearTimeout(this.healTimer);
+        this.healTimer = null;
+    }
+};
+
+DiscoverProviderHealer.prototype._run = function _run() {
+    var self = this;
+
+    if (self.isStopped) {
+        return;
+    }
+
+    if (Math.random() < probability()) {
+        self.heal(function onHeal() {
+            scheduleNext();
```

Comment:
Any reason, other than readability, for why this can't be `self.heal(scheduleNext)`?